src/runtime/pythonengine.cs

```diff
@@ -64,7 +67,17 @@ public static string ProgramName
 
                 return result ?? "";
             }
-            set { Runtime.Py_SetProgramName(value); }
+            set
+            {
+                if (_programName != IntPtr.Zero)
```

Comment:
What would you replace it with? Without it, we'd end up trying to `dealloc` null